content/exchange/artifacts/Generic.Detection.WebShells.yaml

```diff
@@ -0,0 +1,778 @@
+
+        import "pe"
+        private rule SentinelLabs_SUPERNOVA
+        {
+            meta:
+                description = "Identifies potential versions of App_Web_logoimagehandler.ashx.b6031896.dll weaponized with SUPERNOVA"
+                date = "2020-12-22"
+                author = "SentinelLabs"
+                source = "https://labs.sentinelone.com/solarwinds-understanding-detecting-the-supernova-webshell-trojan/"
+                
+            strings:
+                $ = "clazz"
+                $ = "codes"
+                $ = "args"
+                $ = "ProcessRequest"
+                $ = "DynamicRun"
+                $ = "get_IsReusable"
+                $ = "logoimagehandler.ashx" wide
+                $ = "SiteNoclogoImage" wide
+                $ = "SitelogoImage" wide
+
+            condition:
+                (uint16(0) == 0x5A4D and uint32(uint32(0x3C)) == 0x00004550 and pe.imports("mscoree.dll")) and all of them
+        }
+
+        rule SolarWindsArtifacts
+        {
+            meta:
+                author = "NSA Cybersecurity"
+                description = "Artifacts common to the SolarWinds compromise."
+
+            condition:
+                APT_Backdoor_MSIL_SUNBURST_1 
+                or APT_Backdoor_MSIL_SUNBURST_2 
+                or APT_Backdoor_MSIL_SUNBURST_3 
+                or APT_Backdoor_MSIL_SUNBURST_4 
+                or APT_Dropper_Raw64_TEARDROP_1 
+                or APT_Dropper_Win64_TEARDROP_2
+                or SentinelLabs_SUPERNOVA
+        }
+
+        rule reGeorg_Variant_Web_shell {
+            meta:
+                description = "Matches the reGeorg variant web shell used by the actors."
+                date = "2021-07-01"
+                author = "National Security Agency"
+                source = "https://media.defense.gov/2021/Jul/01/2002753896/-1/-1/1/CSA_GRU_GLOBAL_BRUTE_FORCE_CAMPAIGN_UOO158036-21.PDF"
+                
+            strings:
+                $pageLanguage = "<%@ Page Language=\"C#\""
+                $obfuscationFunction = "StrTr"
+                $target = "target_str"
+                $IPcomms = "System.Net.IPEndPoint"
+                $addHeader = "Response.AddHeader"
+                $socket = "Socket"
+                
+            condition:
+                5 of them
+        }
+
+sources:
+  - name: YaraHits
+    query: |
+      LET webroot = SELECT * FROM switch(
+        windows={SELECT WindowsWebRoot AS Dir FROM info() WHERE OS = "windows"},
+        linux={SELECT LinuxWebRoot AS Dir FROM info() WHERE OS = "linux"}
+      ) 
+      LET webroot_dir <= webroot[0].Dir
+      Select * FROM Artifact.Generic.Detection.Yara.Glob(PathGlob=webroot_dir, YaraRule=YaraRule, DateAfter=DateAfter)
+  - name: WindowsProcessCreation
+    precondition:
+      SELECT OS From info() where OS = 'windows'
+    query: |
+      SELECT * FROM Artifact.Windows.EventLogs.Evtx(EvtxGlob='%SystemRoot%\\System32\\winevt\\Logs\\Security\.evtx', IDRegex="4688")
```

Comment:
Might consider using Velociraptor sigma capability released version 0.7.1 to help with this part.
https://github.com/Velocidex/velociraptor-sigma-rules

Should help you cover evtx (security, sysmon) and using process tracker sources.